R/popover.R

```diff
@@ -0,0 +1,161 @@
+#' Add a popover to a UI element
+#'
+#' Display additional information when clicking on a UI element (typically a
+#' button).
+#'
+#' @param trigger The UI element to serve as the popover trigger (typically a
+#'   [shiny::actionButton()] or similar). If `trigger` renders as multiple HTML
+#'   elements (e.g., it's a `tagList()`), the last HTML element is used for the
+#'   trigger. If the `trigger` should contain all of those elements, wrap the
+#'   object in a [div()] or [span()].
+#' @param ... UI elements for the popover's body. Character strings are
+#'   [automatically escaped][htmlEscape()] unless marked as [HTML()].
+#' @param title A title (header) for the popover.
+#' @param id A character string. Required to re-actively respond to the
+#'   visibility of the popover (via the `input[[id]]` value) and/or update the
+#'   visibility/contents of the popover.
+#' @param placement The placement of the popover relative to its trigger.
+#' @param close_button Whether to include a close button in the popover.
+#' @param options A list of additional
+#'   [options](https://getbootstrap.com/docs/5.3/components/popovers/#options).
+#'
+#' @section Closing popovers:
+#'
+#'   In addition to clicking the `close_button`, popovers can be closed by
+#'   pressing the Esc/Space key when the popover (and/or its trigger) is
+#'   focused.
+#'
+#' @section Theming/Styling:
+#'
+#'   Like other bslib components, popovers can be themed by supplying [relevant
+#'   theming
+#'   variables](https://rstudio.github.io/bslib/articles/bs5-variables.html#popover-bg)
+#'   to [bs_theme()], which effects styling of every popover on the page. To
+#'   style a _specific_ popover differently from other popovers, utilize the
+#'   `customClass` option:
+#'
+#'   ```
+#'   popover(
+#'     "Trigger", "Popover message",
+#'     options = list(customClass = "my-pop")
+#'   )
+#'   ```
+#'
+#'   And then add relevant rules to [bs_theme()] via [bs_add_rules()]:
+#'
+#'   ```
+#'   bs_theme() |> bs_add_rules(".my-pop { max-width: none; }")
+#'   ```
```

Comment:
We should demo using BS CSS variables in this example

Reply:
I'm not sure the extra complexity of explaining that (and the added BS dependency) is worth the effort?